testing/demo/pkg/transfer/update_groth16_client.go

```diff
@@ -71,37 +73,78 @@ func updateGroth16LightClient(evmTransferBlockNumber uint64) error {
 	return nil
 }
 
+// DecodePublicValues decodes the public values from the Blevm proof aggregator.
+// The public values are encoded using bincode serialization. They're encoded in fixed order.
+// All of the fields are fixed size bytes32 and are encoded in little endian format.
+// Celestia header hashes is not a fixed size array therefore it is encoded
+// as a u64 length prefix(bytes8) followed by the hashes.
+func DecodePublicValues(data []byte) (*groth16.BlevmAggOutput, error) {
+	buf := bytes.NewBuffer(data)
+	output := &groth16.BlevmAggOutput{}
+
+	// Read fixed-size fields they should be 32 bytes each
+	if err := binary.Read(buf, binary.LittleEndian, &output.NewestHeaderHash); err != nil {
+		return nil, fmt.Errorf("read newest header hash: %w", err)
+	}
+
+	if err := binary.Read(buf, binary.LittleEndian, &output.OldestHeaderHash); err != nil {
+		return nil, fmt.Errorf("read oldest header hash: %w", err)
+	}
+
+	// Celestia header hashes are of variable length but bincode serialization
+	// pefixes them with a u64(8 bytes) length. We slice the 32 bytes times the length.
+	celestiaHeaderHashesLength := binary.LittleEndian.Uint64(data[64 : 64+8])
```

Comment:
[optional] it would be nice to check if data is at least of length 64 + 8 prior to this line and throw an error saying something like "data is not large enough to include the celestia header hash length".